scripts/publish.sh

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env bash
+#
+# This script is copied from `https://github.com/paritytech/jsonrpsee` with some minor tweaks.
+# Add `--dry-run` and/or `--allow-dirty` to your command line to test things before publication.
+
+set -eu
+
+# Find publishable crates by running something like this below and figure out the topological order:
+# $ find . -iname 'Cargo.toml' | xargs grep "publish\ *=\ *true" | grep "^.*\.toml" -o
+ORDER=(metadata codegen macro subxt cli)
+
+function read_toml () {
+	NAME=""
+	VERSION=""
+	# There can be multiple "name = ..." lines, so make sure to only catch the first one after the "[package]" section
+	NAME=$(grep -Pzo "\[package\]\\n([^name].*\\n)*name\ *=\ *\"(.*)\"" ./Cargo.toml | tr -d '\000' | grep -a name | sed -e 's/.*"\(.*\)"/\1/')
```

Comment:
This was specifically done because there's a `name = ...` section under `[[bench]]` that this script caught, so I tried to apply the heuristic outlined in the comment to filter out any unwanted names.

Reply:
it is not allowed to have multiple `name` fields in a package:

```toml
[package]
name = "foo"
fields = "..."

# package name = "foo" ends here 
[dependency]
bar = "1"
```

I guess the regex would be even more complicated than that and it might even be better to re-write in rust

Reply:
I meant multiple `name = ...` lines in the file in general, not in the package section. I rephrased that, hopefully it's more clear now